src/fsharp/FSharp.Core/prim-types.fs

```diff
@@ -478,6 +478,11 @@ namespace Microsoft.FSharp.Core
         let inline isinstPrim<'T>(x:obj) = (# "isinst !0" type ('T) x : obj #)
         let inline castclassPrim<'T>(x:obj) = (# "castclass !0" type ('T) x : 'T #)
         let inline notnullPrim<'T when 'T : not struct>(x:'T) = (# "ldnull cgt.un" x : bool #)
+           
+        let inline internal notnullFast (x: ^T) = 
```

Comment:
@dsyme I described the issue at https://github.com/Microsoft/visualfsharp/issues/1231#issuecomment-222929038

I tried to look deeper, but

![tiger](https://cloud.githubusercontent.com/assets/57396/15709846/36fc7bbc-2807-11e6-8bb8-9d49b2dd5feb.gif)